### PR TITLE
Update app initialization doc

### PR DESCRIPTION
**Pull request by konstantin-msft**

### CODE REVIEW

File: lib/msal-browser/docs/initialization.md [hunk 1]
```diff
@@ -15,7 +15,12 @@ In this document:
 
 ## Initializing the PublicClientApplication object
 
-In order to use MSAL.js, you need to instantiate a `PublicClientApplication` object. You must provide the `client id` (`appId`) of your application. 
+In order to use MSAL.js, you need to instantiate a `PublicClientApplication` object. You must provide the `client id` (`appId`) of your application.
+
+### Option 1
+
+Instantiate an application object and initialize it afterwards.
```
Comment (tnorling) [inline lib/msal-browser/docs/initialization.md]:
```suggestion
Instantiate a `PublicClientApplication` object and initialize it afterwards. The `initialize` function is asynchronous and must resolve before invoking other MSAL.js APIs.
```

File: lib/msal-browser/docs/initialization.md [hunk 2]
```diff
@@ -26,6 +31,23 @@ const msalConfig = {
 };
 
 const msalInstance = new msal.PublicClientApplication(msalConfig);
+await msalInstance.initialize();
+```
+
+### Option 2
+
+Invoke a static method to get initialized application object.
```
Comment (tnorling) [inline lib/msal-browser/docs/initialization.md]:
```suggestion
Invoke the `createPublicClientApplication` static method which returns an initialized `PublicClientApplication` object. Note that this function is asynchronous.
```

**Review by tnorling — APPROVED** [review]
A couple suggestions to add clarity. Would also be good to start a migration doc where we can enumerate all the changes between v2 and v3.

File: lib/msal-browser/docs/initialization.md (hunk 1 above)
Comment (konstantin-msft) [reply lib/msal-browser/docs/initialization.md]: Updated

File: lib/msal-browser/docs/initialization.md (hunk 2 above)
Comment (konstantin-msft) [reply lib/msal-browser/docs/initialization.md]: Updated

**Comment by konstantin-msft** [comment]
> A couple suggestions to add clarity. Would also be good to start a migration doc where we can enumerate all the changes between v2 and v3.

Good idea. I will add a migration doc in a separate PR. Thanks for review!

**Review by derisen — APPROVED** [review]

**Pull request merged**